### DISCUSSION TITLE
Creating Extensions with Maven

### DISCUSSION THREAD

**Discussion by alvin562** [open-telemetry/opentelemetry-java-instrumentation#6643] [Q&A]
I am trying to create an extension to exclude health check spans. I am following the sample extensions provided https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/examples/extension/src/main/java/com/example/javaagent/DemoSampler.java and noticed it is a gradle project.  My question is does it matter if we use maven instead? Thanks.

**Comment by mateuszrzeszutek** [comment]
Hey @alvin562 ,
Unless you want to introduce a custom [`InstrumentationModule`](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/extension/instrumentation/InstrumentationModule.java) it's perfectly fine to use maven. The reason why we use gradle there is that we have a custom gradle plugin that does some bytecode magic during compilation, but that only affects javaagent instrumentation code (and, we use gradle everywhere in this project, so consistency's another reason). If you only want to customize the SDK and do not care about adding custom bytecode instrumentation then maven's perfectly fine.
